### PR TITLE
#283 Support for server-drive event index (stream version)

### PR DESCRIPTION
**Pull request by alexeyzimarev**
Check my attempt to solve it. I am not sure about the API change impact. It only concerns providers since call results were not used anyway, so all usages are not affected.

### CODE REVIEW

**Comment by tomliversidge** [comment]
Looks ok to me!

**Comment by raskolnikoov** [comment]
Im gonna have a look to just in a sec

File: src/PersistenceProviders/Proto.Persistence.Sqlite/SqliteProvider.cs [hunk 1]
```diff
@@ -69,7 +69,7 @@ public SqliteProvider(string datasource = "states.db")
             }
         }
 
-        public Task GetEventsAsync(string actorName, long indexStart, long indexEnd, Action<object> callback)
+        public async Task<long> GetEventsAsync(string actorName, long indexStart, long indexEnd, Action<object> callback)
```
Comment (raskolnikoov) [inline src/PersistenceProviders/Proto.Persistence.Sqlite/SqliteProvider.cs]: remove async

File: src/PersistenceProviders/Proto.Persistence.Sqlite/SqliteProvider.cs [hunk 2]
```diff
@@ -84,14 +84,13 @@ public Task GetEventsAsync(string actorName, long indexStart, long indexEnd, Act
                     {
                         callback(JsonConvert.DeserializeObject<object>(item.EventData, new JsonSerializerSettings { TypeNameHandling = TypeNameHandling.Auto }));
                     }
+                    return items.Any() ? items.LastOrDefault().EventIndex : -1;
```
Comment (raskolnikoov) [inline src/PersistenceProviders/Proto.Persistence.Sqlite/SqliteProvider.cs]: should looks like this instead?

return Task.FromResult(items.Any() ? items.LastOrDefault().EventIndex : -1);

Comment (alexeyzimarev) [inline src/PersistenceProviders/Proto.Persistence.Sqlite/SqliteProvider.cs]: Yeah, if you remove `async`...

**Pull request merged**